numpyro/distributions/constraints.py

```diff
@@ -697,6 +698,30 @@ def feasible_like(self, prototype):
         return jax.numpy.full_like(prototype, prototype.shape[-1] ** (-0.5))
 
 
+class _ZeroSum(Constraint):
+    def __init__(self, event_dim=1):
+        self.event_dim = event_dim
+        super().__init__()
+
+    def __call__(self, x):
+        jnp = np if isinstance(x, (np.ndarray, np.generic)) else jax.numpy
+        zerosum_true = True
+        for dim in range(-self.event_dim, 0):
+            zerosum_true = zerosum_true & jnp.allclose(
```

Comment:
Good callout, the tolerance seems like it can be much more strict than it was. I actually decided to follow the first example and use `tol = jnp.finfo(x.dtype).eps * x.shape[-1] * 10` as the `atol` and removed the rtol - do you think thats sufficient?